### PR TITLE
Enable Indexing by int-Compatible type

### PR DESCRIPTION
**Pull request by justusschock**
Fixes

```python
  File "/work/scratch/schock/conda/envs/oai/lib/python3.8/site-packages/torch/utils/data/_utils/worker.py", line 287, in _worker_loop
    data = fetcher.fetch(index)
  File "/work/scratch/schock/conda/envs/oai/lib/python3.8/site-packages/torch/utils/data/_utils/fetch.py", line 44, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/work/scratch/schock/conda/envs/oai/lib/python3.8/site-packages/torch/utils/data/_utils/fetch.py", line 44, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/work/scratch/schock/conda/envs/oai/lib/python3.8/site-packages/torch/utils/data/dataset.py", line 311, in __getitem__
    return self.dataset[self.indices[idx]]
  File "/work/scratch/schock/conda/envs/oai/lib/python3.8/site-packages/torchio/data/dataset.py", line 77, in __getitem__
    raise ValueError(f'Index "{index}" must be int, not {type(index)}')
ValueError: Index "382" must be int, not <class 'torch.Tensor'>
```

**Description**
Latest torch samplers (and thereby dataloaders) index with Tensors.
This PR enhances datasets to also work with int-compatible types like scalar torch Tensors or numpy arrays.

**Checklist**

- [x] I have read the [`CONTRIBUTING`](https://github.com/fepegar/torchio/blob/master/CONTRIBUTING.rst) docs and have a developer setup (especially important are `pre-commit`and `pytest`)
- [x] Non-breaking change (would not break existing functionality)
- [ ] Breaking change (would cause existing functionality to change)
- [x] Tests added or modified to cover the changes
- [ ] Integration tests passed locally by running `pytest`
- [ ] In-line docstrings updated
- [ ] Documentation updated, tested running `make html` inside the `docs/` folder
- [x] This pull request is ready to be reviewed
- [ ] If the PR is ready and there are multiple commits, I have [squashed them and force-pushed](https://www.w3docs.com/snippets/git/how-to-combine-multiple-commits-into-one-with-3-steps.html#force-pushing-commits-7)

### CODE REVIEW

File: tests/data/test_subjects_dataset.py [hunk 1]
```diff
@@ -5,6 +5,10 @@
 
 
 class TestSubjectsDataset(TorchioTestCase):
+
```
Comment (fepegar) [inline tests/data/test_subjects_dataset.py:8]: Could you please remove this line as well as line 76 from the other file?

**Review by fepegar — CHANGES_REQUESTED** [review]
Great! Thanks, @justusschock. Just requested some minor edits.

File: tests/data/test_subjects_dataset.py (hunk 1 above)
Comment (justusschock) [reply tests/data/test_subjects_dataset.py]: done :)

**Review by fepegar — APPROVED** [review]

**Comment by fepegar** [comment]
Thanks again, @justusschock.

@all-contributors please add @justusschock for bug

**Pull request merged**